missing onnxruntime_c_api.h  : firewall issue and no warning/error

Hello Michele, first of all : thank you for your work on onnx2fmu!

I face a issue  on Windows 11/VisualStudio2022. I try to run Exemple 1.
Notebook `generate-example-model.ipynb` ran like a charm and produced : `example1.onnx` and `example1Description.json` as exepected.
But when I run `onnx2fmu build .\example1.onnx .\example1Description.json out` I face:
```
...
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- ONNX Runtime archive already exists at C:\Users\xxxxxxx\AppData\Local/onnxruntime/onnxruntime-win-x64-1.20.1.zip
CMake Error at CMakeLists.txt:189 (message):
  ONNX Runtime extraction failed.  File not found:                                                                                                               
  C:/Users/a067220/repo/onnx/example1/out/build/onnxruntime-win-x64-1.20.1/include/onnxruntime_c_api.h                                                           
                                                                                                                                                                 
                                                                                                                                                                 
-- Configuring incomplete, errors occurred!
``` 

Note: `AppData\Local/onnxruntime/onnxruntime-win-x64-1.20.1.zip` exists but is empty.


--EDIT--
After a second look: it seems the download of runtime failed due to network company configuration. After manual download, I succeed to build the FMU.

It would be great to keep the runtime with python package :-)

Kind regards

--- a/onnx2fmu/CMakeLists.txt
+++ b/onnx2fmu/CMakeLists.txt
@@ -179,6 +179,23 @@ else ()
     message(STATUS "ONNX Runtime archive already exists at ${ONNXRUNTIME_ARCHIVE_PATH}")
 endif ()
 
+# A firewall or proxy may silently produce a zero-byte file instead of the real archive.
+# Detect this early, remove the empty file (so the next run re-downloads), and give
+# the user actionable instructions.
+file(SIZE ${ONNXRUNTIME_ARCHIVE_PATH} ONNXRUNTIME_ARCHIVE_SIZE)
+if (ONNXRUNTIME_ARCHIVE_SIZE EQUAL 0)
+    file(REMOVE ${ONNXRUNTIME_ARCHIVE_PATH})
+    message(FATAL_ERROR
+        "ONNX Runtime archive is empty (0 bytes): ${ONNXRUNTIME_ARCHIVE_PATH}\n"
+        "This is usually caused by a firewall or proxy blocking the download.\n"
+        "Please download the archive manually from:\n"
+        "  ${ONNXRUNTIME_URL}\n"
+        "save it to:\n"
+        "  ${ONNXRUNTIME_ARCHIVE_PATH}\n"
+        "and re-run CMake."
+    )
+endif ()
+
 # Extract the ONNX Runtime release
 execute_process(
     COMMAND ${CMAKE_COMMAND} -E tar xzf ${ONNXRUNTIME_ARCHIVE_PATH}
@@ -189,7 +206,15 @@ execute_process(
 if (EXISTS "${CMAKE_CURRENT_BINARY_DIR}/${ONNXRUNTIME}/include/onnxruntime_c_api.h")
   message(STATUS "ONNX Runtime extraction successful.")
 else ()
-  message(FATAL_ERROR "ONNX Runtime extraction failed. File not found: ${CMAKE_CURRENT_BINARY_DIR}/${ONNXRUNTIME}/include/onnxruntime_c_api.h")
+  message(FATAL_ERROR
+      "ONNX Runtime extraction failed — "
+      "${CMAKE_CURRENT_BINARY_DIR}/${ONNXRUNTIME}/include/onnxruntime_c_api.h not found.\n"
+      "The archive at ${ONNXRUNTIME_ARCHIVE_PATH} may be corrupt or incomplete.\n"
+      "Delete it and re-run CMake, or download manually from:\n"
+      "  ${ONNXRUNTIME_URL}\n"
+      "and save to:\n"
+      "  ${ONNXRUNTIME_ARCHIVE_PATH}"
+  )
 endif ()
 
 # Include the onnxruntime directory like in https://github.com/microsoft/onnxruntime-inference-examples/blob/main/c_cxx/CMakeLists.txt

